internal/controllers/app_controller.go

```diff
@@ -270,12 +290,245 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App, logger
 		}
 	}
 
+	if len(app.Spec.Deployments) > 0 {
+		// use latest deployment and watch events for each process
+		latestDeployment := app.Spec.Deployments[len(app.Spec.Deployments)-1]
+		for _, process := range latestDeployment.Processes {
+			var dep appsv1.Deployment
+			if err := r.Get(ctx, client.ObjectKey{
+				Namespace: framework.Spec.NamespaceName,
+				Name:      fmt.Sprintf("%s-%s-%d", app.GetName(), process.Name, latestDeployment.Version),
+			}, &dep); err != nil {
+				return reconcileResult{
+					status:  v1.ConditionFalse,
+					message: fmt.Sprintf("failed to get deployment: %v", err),
+				}
+			}
+			err = r.watchDeployEvents(ctx, app, framework.Spec.NamespaceName, &dep, &process, r.Recorder)
+			if err != nil {
+				return reconcileResult{
+					status:  v1.ConditionFalse,
+					message: fmt.Sprintf("failed to get deploy events: %v", err),
+				}
+			}
+		}
+	}
+
 	return reconcileResult{
 		framework: ref,
 		status:    v1.ConditionTrue,
 	}
 }
 
+// watchDeployEvents watches a namespace for events and, after a deployment has started updating, records events
+// with updated deployment status and/or healthcheck and timeout failures
+func (r *AppReconciler) watchDeployEvents(ctx context.Context, app *ketchv1.App, namespace string, dep *appsv1.Deployment, process *ketchv1.ProcessSpec, recorder record.EventRecorder) error {
+	config, err := GetRESTConfig()
+	if err != nil {
+		return err
+	}
+	cli, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	opts := metav1.ListOptions{
+		FieldSelector: "involvedObject.kind=Pod",
```

Comment:
should it be for the given app?

Reply:
I'm not sure there is a way to filter Pod events by App since events don't have labels. `isDeploymentEvent` filters them by app name prefix before emitting them as App Events.